packages/node_modules/@ciscospark/plugin-memberships/src/memberships.js

```diff
@@ -144,70 +122,82 @@ const Memberships = SparkPlugin.extend({
+        sdkEvent.data.roomType =
+          activity.target.tags.includes(SDK_EVENT.FROM_WEBEX.ACTIVITY_TAG.ONE_ON_ONE) ?
+            SDK_EVENT.TO_APP.SPACE_TYPE.DIRECT :
+            SDK_EVENT.TO_APP.SPACE_TYPE.GROUP;
+        sdkEvent.data.isRoomHidden = false; // any activity unhides a space.
+      }
+      else {
+        sdkEvent.data.roomId =
+        constructHydraId(hydraTypes.ROOM, activity.object.id);
+        // currently hidden attribute is only set on 1-1
+        sdkEvent.data.roomType = SDK_EVENT.TO_APP.SPACE_TYPE.DIRECT;
+        sdkEvent.data.isRoomHidden = true;
+      }
+      if (activity.verb !== SDK_EVENT.FROM_WEBEX.ACTIVITY_VERB.ACKNOWLEDGE) {
+        if ((activity.object.roomProperties) && (activity.object.roomProperties.isModerator)) {
+          sdkEvent.data.isModerator = true;
+        }
+        else {
+          sdkEvent.data.isModerator = false;
+        }
+      }
+      // This is deprecated but still sent in the webhooks
+      // We won't send it for our new SDK events
+      // sdkEvent.data.isMonitor = false;
+
+      if (activity.verb === SDK_EVENT.FROM_WEBEX.ACTIVITY_VERB.ACKNOWLEDGE) {
+        // For a read receipt the person is the "actor" or the one who did the reading
+        member = SDK_EVENT.FROM_WEBEX.ACTIVITY_FIELD.ACTOR;
+        // The space with the read message is the "target"
+        space = SDK_EVENT.FROM_WEBEX.ACTIVITY_FIELD.TARGET;
+        // And the "object" is the message that was last seen
+        sdkEvent.data.lastSeenId =
+          constructHydraId(hydraTypes.MESSAGE, activity.object.id);
+      }
+      else if (activity.verb === SDK_EVENT.FROM_WEBEX.ACTIVITY_VERB.HIDE) {
+        // For a hide activity the person is also the "actor"
+        member = SDK_EVENT.FROM_WEBEX.ACTIVITY_FIELD.ACTOR;
+        // But the space is now the "object"
+        space = SDK_EVENT.FROM_WEBEX.ACTIVITY_FIELD.OBJECT;
+      }
+      else {
+        // For most memberships events the person is the 'object"
+        member = SDK_EVENT.FROM_WEBEX.ACTIVITY_FIELD.OBJECT;
+        // and the space is the "target"
+        space = SDK_EVENT.FROM_WEBEX.ACTIVITY_FIELD.TARGET;
+      }
+
+      sdkEvent.data.id = constructHydraId(hydraTypes.MEMBERSHIP,
+        `${activity[member].entryUUID}:${activity[space].id}`);
+      sdkEvent.data.personId =
+        constructHydraId(hydraTypes.PEOPLE, activity[member].entryUUID);
+      sdkEvent.data.personEmail =
+        activity[member].emailAddress || activity[member].entryEmail;
+      sdkEvent.data.personDisplayName = activity[member].displayName;
+      sdkEvent.data.personOrgId =
+        constructHydraId(hydraTypes.ORGANIZATION, activity[member].orgId);
+
+      return sdkEvent;
     }
-    if ((!activity[member]) || (!activity[space])) {
-      console.error('Unable to generate SDK event ' +
-        'from mercury socket activity');
+    catch (e) {
+      console.error(`Unable to generate SDK event from mercury \
```

Comment:
Let's use the `logger` instead of logging directly to console, e.g., `this.spark.logger.error(...`

Reply:
Agree, see commit 41ff1a54e925f04e1588abc4d53b969c05b123cc